### ISSUE TITLE
Font files are classified as Scala or HolyC

### ISSUE THREAD

**Issue by aBARICHELLO** [github/linguist#4870]
Font source files included in repositories are being misidentified as Objective-J/HolyC/Scala.

### URL of the affected repository:
https://github.com/ValveSoftware/Proton/tree/proton_5.0
https://github.com/adobe-fonts/source-han-mono

### Expected language:
none

### Detected language:
Scala/HolyC/SuperCollider/etc

**Comment by lildude** [comment]
As you've already found, this is because of the `.sc` extension which is only associated with those three languages and no font "languages" or otherwise explicitly ignored.  As a result, things fall through to the [heuristic](https://github.com/github/linguist/blob/16507700cdf46a78368a571c04d1bc601f62947d/lib/linguist/heuristics.yml#L437-L442) and then on to the classifier, but given Linguist doesn't know anything about the font "language" you're expecting it to find, it'll never find it.

As Linguist relies upon community contributions to address such things, we'd welcome a PR that either adds the language or ignores the files.

**Comment by lildude** [comment]
Oh, this applies to the other extensions too.

**Comment by aeikum** [comment]
A month is a long time? Yes of course it's still an issue. Come on, bot. Am I going to have to do this every month?

**Comment by lildude** [comment]
> A month is a long time? Yes of course it's still an issue. Come on, bot. Am I going to have to do this every month?

Yes, or you can submit a PR to add support 😉, after all Linguist relies almost exclusively on community contributions.

**Comment by aeikum** [comment]
https://github.com/github/linguist/issues/4870#issuecomment-650159950

**Comment by aeikum** [comment]
Asdf

**Comment by Alhadis** [comment]
@aeikum Could you enlighten us on what `.SC` files are? I googled `SC font file" but it (naturally) brings up results for `small-cap` variants of actual typefaces.

**Comment by aeikum** [comment]
I believe they are variants of the font files for different languages. So they're different file types, despite the same final extension. See here: https://github.com/adobe-fonts/source-han-mono/tree/master/Heavy/OTC

J - Japanese, K - Korean, SC - Simplified Chinese, TC - Traditional Chinese. Not sure about HC.

**Comment by Alhadis** [comment]
> SC - Simplified Chinese, TC - Traditional Chinese. Not sure about HC

Hong Kong and Taiwan have different variations of traditional Chinese, so TC and HC probably stand for "Taiwan Chinese" and "Hong Kong Chinese", respectively.

Anyway. From what I see in [`adobe-han-mono/Heavy/OTC`](https://github.com/adobe-fonts/source-han-mono/tree/9a10a4fe9797b9777dd9b77cd093e8f76dd55e1d/Heavy/OTC), the `.HC` files are a mix of PostScript (Type 1 fonts, essentially specialised PostScript programs), OpenType feature [definitions](https://github.com/adobe-fonts/source-han-mono/blob/0381f376e6fdf00d3d463f5a371bcec4b1b69e23/Heavy/OTC/cidfontinfo.OTC.SC), and some ad hoc-looking format for [CID font metadata](https://github.com/adobe-fonts/source-han-mono/blob/0381f376e6fdf00d3d463f5a371bcec4b1b69e23/Heavy/OTC/cidfontinfo.OTC.J). Since most of these files are too large to be indexed or displayed on GitHub, it's gonna be difficult to accurately determine how many repositories use `.HC` and `.TC` as file extensions (see [`CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language) if you're unsure why that's relevant).

@aBARICHELLO Aside from the two repositories you linked to, how many others have you encountered where this is obviously an issue?

**Comment by smola** [comment]
Note that there was quite some discussion about font files: https://github.com/github/linguist/issues/2516

**Comment by Alhadis** [comment]
That issue's from early/mid 2015. Barely any of it is relevant anymore — several missing font formats were added by yours truly, and being the resident font-nerd, I'm always eager to add support for a font format. 😉

**Comment by aBARICHELLO** [comment]
> @aBARICHELLO Aside from the two repositories you linked to, how many others have you encountered where this is obviously an issue?
[Advanced search](https://github.com/search?q=.SC+extension%3A.SC&type=Code&ref=advsearch&l=&l=) returned hundreds of results for .SC files being identified as Scala. .HC is harder to find.

**Comment by Alhadis** [comment]
Well, if you or anybody else is interested, [here's an (unsorted) stash](https://github.com/Alhadis/Silos/tree/ffec3870d28b3984cee3533fb1f2f7712943af86/files) of `.sc` files harvested from search results. They don't appear to have much (or anything) in common... 😕

**Comment by aeikum** [comment]
Yup
